core/src/main/java/org/apache/gravitino/catalog/SchemaNormalizeDispatcher.java

```diff
@@ -33,26 +35,27 @@
 import org.apache.gravitino.exceptions.SchemaAlreadyExistsException;
 
 public class SchemaNormalizeDispatcher implements SchemaDispatcher {
+  private final CatalogManager catalogManager;
+  private final SchemaDispatcher dispatcher;
 
-  private final SchemaOperationDispatcher dispatcher;
-
-  public SchemaNormalizeDispatcher(SchemaOperationDispatcher dispatcher) {
+  public SchemaNormalizeDispatcher(SchemaDispatcher dispatcher, CatalogManager catalogManager) {
```

Comment:
The key to refactoring is here, replacing `SchemaOperationDispatcher` with `SchemaDispatcher`. 

In the decorator design pattern, we should depend on interface `SchemaDispatcher` instead of the concrete implementation class `SchemaOperationDispatcher`.
@jerryshao